File: Backend-SpringBoot/FoodApp/src/main/java/com/tientien/foodapp/address/entity/Address.java
package com.tientien.foodapp.address.entity;

import com.tientien.foodapp.common.entity.BaseEntity;
import com.tientien.foodapp.user.entity.User;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "addresses")
@Getter
@Setter
public class Address extends BaseEntity {

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private String fullName;

    @Column(nullable = false)
    private String phoneNumber;

    @Column(nullable = false)
    private String addressLine;

    private String city;
    private String district;
    private String ward;

    private Double latitude;
    private Double longitude;

    @Column(nullable = false)
    private Boolean isDefault = false;
}

File: Frontend_Flutter/lib/models/address.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'address.freezed.dart';
part 'address.g.dart';

@freezed
class Address with _$Address {
  const factory Address({
    int? id,
    int? userId,
    String? fullName,
    String? phoneNumber,
    String? addressLine,
    String? city,
    String? district,
    String? ward,
    double? latitude,
    double? longitude,
    bool? isDefault,
    String? createdAt,
  }) = _Address;

  factory Address.fromJson(Map<String, dynamic> json) =>
      _$AddressFromJson(json);
}
